app/lib/test_pals_mock_ldap.rb

```diff
@@ -0,0 +1,58 @@
+# This code mocks the LDAP services that provide authentication to Student Insights
+# by implementing the `bind` method. It uses a dummy password defined in ENV
+# instead of real passwords.
+
+# We will consume this mock service in three places: (1) the test suite, (2) the
+# local development environment, (3) our demo site.
+
+# The advantage to mocking an LDAP service in this way is that developers
+# will exercise the code in our custom LDAP Devise strategy constantly,
+# instead of relying on a different auth mechanism locally and only exercising
+# LDAP-related code in production.
+
+class TestPalsMockLDAP
+
+  def initialize(options)
+    @options = options
+    @email = options[:auth][:username]
+    @password = options[:auth][:password]
+  end
+
+  def bind
+    raise 'LDAP error' unless ::EnvironmentVariable.is_true('USE_TEST_PALS_LDAP')
```

Comment:
Let's make this error message explicit, saying that a method on the fake class was called when the env variable wasn't set.  Otherwise seeing "LDAP error" in the logs might make it seem like an LDAP server was down.